### PR TITLE
Downloadable and Virtual support on backend 

### PR DESCRIPTION
**Pull request by serhii-balko**
### Description
Downloadable products can only be assigned to default source
Virtual products can only be assigned to default source

### Fixed Issues (if relevant)
1. magento-engcom/msi#678: Downloadable support on backend 
2. magento-engcom/msi#679: Virtual support on backend 

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventoryCatalog/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableVirtualTypePlugin.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+namespace Magento\InventoryCatalog\Plugin\InventoryConfiguration\IsSourceItemsAllowedForProductType;
+
+use Magento\Catalog\Model\Product\Type;
+use Magento\InventoryConfiguration\Model\IsSourceItemsAllowedForProductTypeInterface;
+
+/**
+ * Disable Source items management for virtual product type
+ */
+class DisableVirtualTypePlugin
```
Comment (naydav) [inline app/code/Magento/InventoryCatalog/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableVirtualTypePlugin.php]: Looks like after last discussion
We need to remove this plugin

File: app/code/Magento/InventoryDownloadable/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableDownloadableTypePlugin.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+namespace Magento\InventoryDownloadable\Plugin\InventoryConfiguration\IsSourceItemsAllowedForProductType;
+
+use Magento\Downloadable\Model\Product\Type;
+use Magento\InventoryConfiguration\Model\IsSourceItemsAllowedForProductTypeInterface;
+
+/**
+ * Disable Source items management for downloadable product type
+ */
+class DisableDownloadableTypePlugin
```
Comment (naydav) [inline app/code/Magento/InventoryDownloadable/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableDownloadableTypePlugin.php]: Looks like after last discussion
We need to remove this plugin

File: app/code/Magento/InventoryDownloadable/composer.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "magento/module-inventory-downloadable",
+    "description": "N/A",
+    "require": {
+        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0",
+        "magento/framework": "100.3.*",
+        "magento/module-downloadable": "100.3.*"
+    },
+    "suggest": {
+        "magento/module-inventory-configuration": "100.0.*"
```
Comment (naydav) [inline app/code/Magento/InventoryDownloadable/composer.json]: Looks like after last discussion
We need to remove this dependency

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryDownloadable/composer.json (hunk 3 above)
Comment (serhii-balko) [reply app/code/Magento/InventoryDownloadable/composer.json]: Maybe i should to remove module InventoryDownloadable?

File: app/code/Magento/InventoryDownloadable/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableDownloadableTypePlugin.php (hunk 2 above)
Comment (serhii-balko) [reply app/code/Magento/InventoryDownloadable/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableDownloadableTypePlugin.php]: done

File: app/code/Magento/InventoryCatalog/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableVirtualTypePlugin.php (hunk 1 above)
Comment (serhii-balko) [reply app/code/Magento/InventoryCatalog/Plugin/InventoryConfiguration/IsSourceItemsAllowedForProductType/DisableVirtualTypePlugin.php]: done

**Comment by naydav** [comment]
Closed due to the decision: Downloadable and Virtual product should work as Simple

**Pull request closed without merging**